dig.go

```diff
@@ -311,6 +329,28 @@ func (c *Container) getProviders(k key) []provider {
 	return providers
 }
 
+func (c *Container) failNodes(params []*dot.Param, id uintptr) {
+	failed := make([]*dot.Result, len(params))
+
+	for i, p := range params {
+		failed[i] = &dot.Result{Node: p.Node}
+	}
+	c.dg.FailNodes(failed, id)
+}
+
+func (c *Container) failGroupNodes(params []*dot.Param, id uintptr) {
+	// Taking first element since there will only be one group node that fails.
+	p := params[0]
```

Comment:
This is fine for now but consider adding a TODO to receive the group name here so that this looks like,

    func (c *Container) failGroupNode(name string, t reflect.Type)

It matches the rest of the containerStore interface.

Reply:
Similar to above, we need the constructor id to determine the node that failed, so we don't end up colouring the entire group. I changed the function to 
```go
func (c *Container) failGroupNode(name string, t reflect.Type, id uintptr)
```